exporter/exporter.go

```diff
@@ -527,7 +528,13 @@ func (e *Exporter) scrapeRedisHost(ch chan<- prometheus.Metric) error {
 	e.registerConstMetricGauge(ch, "exporter_last_scrape_connect_time_seconds", connectTookSeconds)
 
 	if err != nil {
-		log.Errorf("Couldn't connect to redis instance")
+		var redactedAddr string
+		if redisURL, _ := url.Parse(e.redisAddr); redisURL != nil {
```

Comment:
Maybe catch the error here instead of checking the result for nil? You could then output the error in the else block in log level debug to add addtl information.